riscv/src/register/tests/read_only_csr.rs

```diff
@@ -0,0 +1,133 @@
+use crate::result::{Error, Result};
+
+read_only_csr! {
+    /// test CSR register type
+    Mtest: 0x000,
+    mask: 0b1111_1111_1111,
+    /// test single-bit field
+    single: 0,
+}
+
+read_only_csr_field! {
+    Mtest,
+    /// multiple single-bit field range
+    multi_range: 1..=3,
+}
+
+read_only_csr_field!(
+    Mtest,
+    /// multi-bit field
+    multi_field: [4:7],
+);
+
+read_only_csr_field!(
+    Mtest,
+    /// multi-bit field
+    field_enum: {
+        /// field enum type with valid field variants
+        MtestFieldEnum {
+            range: [7:11],
+            default: Field1,
+            Field1 = 1,
+            Field2 = 2,
+            Field3 = 3,
+            Field4 = 15,
+        }
+    }
+);
```

Comment:
You can see how much the `paste` crate simplifies macro invocation, it is substantial.

The savings are even greater for the `read_write_csr*` variants.